libass/ass_parse.c

```diff
@@ -194,11 +195,28 @@ inline void change_alpha(uint32_t *var, int32_t new, double pwr)
  * \param a first value
  * \param b second value
  * \return result of multiplication
- * Parameters and result are limited by 0xFF.
+ * At least one of the parameters must be less than or equal to 0xFF.
+ * The result is less than or equal to max(a, b, 0xFF).
  */
 inline uint32_t mult_alpha(uint32_t a, uint32_t b)
 {
-    return 0xFF - (0xFF - a) * (0xFF - b) / 0xFF;
+    return a - ((uint64_t) a * b + 0x7F) / 0xFF + b;
```

Comment:
To be clear, this expression is not an exact copy of VSFilter’s, but I’ve verified that it produces the same value. I’ve run exhaustive tests on all `a` from 0 to 255 and `b` in ranges around zero, `INT32_MAX / 255` and `INT32_MAX`.